packages/burner-wallet-connector/src/BurnerWalletConnector.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  type AbiMap,
+  type Asset,
+  type ConnectorMetadata,
+  FuelConnector,
+  FuelConnectorEventTypes,
+  type JsonAbi,
+  type Network,
+  Provider,
+  type StorageAbstract,
+  type TransactionRequestLike,
+  type Version,
+  Wallet,
+  type WalletUnlocked,
+} from 'fuels';
+import {
+  BETA_5_URL,
+  BURNER_WALLET_CONNECTED,
+  BURNER_WALLET_ICON,
+  BURNER_WALLET_PRIVATE_KEY,
+  WINDOW,
+} from './constants';
+import type { BurnerWalletConfig } from './types';
+
+export class BurnerWalletConnector extends FuelConnector {
+  name = 'Burner Wallet';
+
+  connected = false;
+  installed = false;
+
+  events = FuelConnectorEventTypes;
+
+  metadata: ConnectorMetadata = {
+    image: BURNER_WALLET_ICON,
+    install: {
+      action: '',
+      description: 'Burner Wallet to connect to Fuel',
+      link: '',
+    },
+  };
+
+  burnerWallet: WalletUnlocked | null = null;
+  burnerWalletProvider: Provider | null = null;
+  burnerWalletPrivateKey: string | null = null;
+
+  private config: BurnerWalletConfig = {};
+  private storage: StorageAbstract;
+
+  constructor(config: BurnerWalletConfig = {}) {
+    super();
+
+    this.config = config;
+    this.storage = this.getStorage(config.storage);
+
+    this.configFuelProvider(config);
+    this.setupBurnerWallet();
+  }
+
+  async configFuelProvider(config: BurnerWalletConfig = {}) {
+    this.config.fuelProvider =
+      config.fuelProvider || Provider.create(BETA_5_URL);
+  }
+
+  async setupBurnerWallet() {
```

Comment:
This methods should be private, and allow to provide only via constructor. If you allow like this it can have a corner case where the configFuelProvider changes but the current instance of the wallet is the same of the old one.